# Multi-level modelling assignment

- The assignment
- Loading packages and reading in data
- Part A
- Part B

## The assignment

#### Multilevel coursework assignment:

Use the “coursework assignment data.ws” in Mlwin to answer the following questions. The data come from a random subsample of the Understanding Society wave 1 survey (for more details, see here: https://www.understandingsociety.ac.uk/documentation) . There are two parts to the assignment (part A and B).

#### The variable names include:

region: Government Office Region

hid: household ID

pid: person ID

age: ranging from 16-96 years

sclfsato: overall satisfaction with life ranging from 1 (completely dissatisfied) to 7 (completely satisfied

nhood_mistrust: neighbourhood mistrust scale ranging from 1 (strongly trust in neighbourhood) to 5 (strongly mistrust 
in neighbourhood) 

urban: rural (0) vs urban (1) neighbourhood

female: man (0) vs woman (1)

hhtenure: housing tenure: owner/mortgaged (1); local authority/housing association rent (2); private rent (3)

hiqual3: highest qualification: Degree or equivalent (1); school level qualification (2); No qualification (3)

worry_crime: Worry about being affected by crime: No (0) vs Yes (1)

Cons, denom: constant terms (1) 

#### Part A: Dependent variable: nhood_mistrust (60 marks)

Fit a multilevel model with nhood_mistrust as the dependent variable. 

- Q1. Provide some summary statistics about the levels in the data. How many units are there at each level (overall N of each level), and how many units are there within levels (N within each level). - 3 marks

- Q2. Specify the generalised equations of a three level multilevel model with random slopes at levels 2 and 3. State the assumptions of the model. - 2 marks

- Q3. Start from the single level null model and add in the household and then region levels. 

A. Display the model coefficients in a table. - 3 marks

B. Does the addition of the household level improve the fit of the model? -2 marks

C. What evidence is there for an improvement in fit? - 5 marks

- Q4: Calculate the VPC at the household and regional levels for the 3 level variance components model. - 3 marks

- Q5. Add in the following explanatory variables in the model (random intercepts model only; no random slopes or coefficients)- age, sclfsato, urban, female, hhtenure, hiqual3. Take out any non-significant associations. 

A. Display the model coefficients in a table. - 2 marks

B. Interpret the coefficients. - 5 marks

- Q6. Add in random slopes for age, at the household and regional level

A. Does this improve the fit of the model?  - 3 marks

B. Take out from the model any non-significant random slope(s) for age

C. Interpret the random slope(s) for age, with the help of graphs. - 10 marks

- Q7. Examine potential interaction effects between age and the other explanatory variables- interpret the interactions using graphs. Is the random slope of age is explained by these interaction terms? - 10 marks

- Q8. After fitting all the explanatory variables, is a three level model still appropriate? - 2 marks

- Q9. Display and interpret the parameters in your final model in a table. - 5 marks

- Q10. Examine residuals at all three levels- are the assumptions of the regression model met? - 5 marks



#### Part B: Dependent variable: worry_crime (20 marks)

Fit a multilevel logistic model with worry_crime as the dependent variable. 

- Q11. Start from the single level null model and add in the household and then GOR levels. Display the model coefficients for the 3 level variance components model  in a table. - 5 marks

- Q12. Is a three level model appropriate? Is a 2 level model appropriate? Is a single level model appropriate? If a 2 level model is appropriate, which two level models (individuals within HH or individuals within GOR)? State your reasons for choosing between a 2 vs 3 level model. - 5 marks

- Q13. Having decided whether a multilevel or single level model is appropriate, add in the explanatory variables from your final model in Part A. Compare the models using a Table. Are the associations with worry_crime similar to the associations with nhood_mistrust?  - 5 marks

- Q14. Take out any non-significant explanatory variables from the model. Interpret the coefficients in your final model in a table.  5 marks

#### NB: 

An additional 20 marks will be assigned for the presentation of tables (10) and figures (10).

The results of this analysis should be presented in a written report of not more than 3000 words. There is no minimum number of words but each question should be answered comprehensively (Not just YES or NO answers, but giving the justification and evidence behind the answers). There is no limit on the number of tables and figures for the report.

Submission deadline: 3pm 14 May 2020
Please submit using Turnitin.
Please enter your ID number on your essay. 
Save your submission file as Student ID number_submissiontitle.doc (eg 1234567_Essay.doc) Enter your Student ID number in the Turnitin title field at the time of submission.

#### Loading packages, reading in data, recoding categorical variables.

- Loading packages

In [1]:
library(tidyverse)
library(fastDummies)
library(broom)
library(lme4)
library(lmerTest)
library(aods3)
#library(sjPlot)
#library(sjmisc)
#library(sjlabelled)

Registered S3 method overwritten by 'rvest':
  method            from
  read_xml.response xml2
── Attaching packages ─────────────────────────────────────── tidyverse 1.2.1 ──
✔ ggplot2 3.1.1       ✔ purrr   0.3.2  
✔ tibble  2.1.1       ✔ dplyr   0.8.0.1
✔ tidyr   0.8.3       ✔ stringr 1.4.0  
✔ readr   1.3.1       ✔ forcats 0.4.0  
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()


ERROR: Error in library(fastDummies): there is no package called ‘fastDummies’


- Remouving NA's

In [3]:
mydataNAs <- read.table(file = "coursework.txt", sep = "," , header = TRUE)
mydata <- drop_na(mydataNAs)

- Creating dummies

In [4]:
mydata <- dummy_cols(mydata, select_columns = c('hhtenure', 'hiqual3'), remove_first_dummy = TRUE)

In [5]:
mydata <- mydata %>% rename(rent_local_auth = hhtenure_2, rent_private = hhtenure_3,
                            school_qual = hiqual3_2, no_qual = hiqual3_3)
colnames(mydata)

[1] "region"          "hid"             "pid"             "age"            
 [5] "sclfsato"        "nhood_mistrust"  "urban"           "female"         
 [9] "hhtenure"        "hiqual3"         "worry_crime"     "cons"           
[13] "denom"           "rent_local_auth" "rent_private"    "school_qual"    
[17] "no_qual"

- Centering age

In [8]:
mydata <- mydata %>% mutate(agenorm = age - 16)

In [12]:
summary(mydata$agenorm)

   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
    0.0    17.0    31.0    31.2    45.0    80.0 

#### Part A

- Q1. Provide some summary statistics about the levels in the data. How many units are there at each level (overall N of each level), and how many units are there within levels (N within each level).

In [5]:
sapply(mydataNAs, function(x) length(unique(x)))

region            hid            pid            age       sclfsato 
            12           4429           4655             80              8 
nhood_mistrust          urban         female       hhtenure        hiqual3 
            25              2              2              4              4 
   worry_crime           cons          denom 
             3              1              1

In [6]:
sapply(mydata, function(x) length(unique(x)))

region             hid             pid             age        sclfsato 
             12            3883            4064              79               7 
 nhood_mistrust           urban          female        hhtenure         hiqual3 
             22               2               2               3               3 
    worry_crime            cons           denom rent_local_auth    rent_private 
              2               1               1               2               2 
    school_qual         no_qual 
              2               2

There are 12 different Government office regions (level 3), 4429 different households (level 2), and 4655 individuals (level 1). The table below provides information for every region on the number of households, individuals, and the mean number of individuals per household.

In [7]:
mydataNAs %>%
    group_by(region) %>% 
    summarise(Number_of_households = n_distinct(hid),
             Number_of_individuals = n(),
             Mean_number_of_individuals_per_household = n()/n_distinct(hid))

region Number_of_households Number_of_individuals
1   1     142                  148                  
2   2     483                  506                  
3   3     333                  353                  
4   4     333                  356                  
5   5     309                  330                  
6   6     350                  364                  
7   7     531                  554                  
8   8     512                  542                  
9   9     360                  374                  
10 10     333                  353                  
11 11     408                  425                  
12 12     335                  350                  
   Mean_number_of_individuals_per_household
1  1.042254                                
2  1.047619                                
3  1.060060                                
4  1.069069                                
5  1.067961                                
6  1.040000                                
7  1.043315                                
8  1.058594                                
9  1.038889                                
10 1.060060                                
11 1.041667                                
12 1.044776

After droping rows with incomplete values there are 12 different Government office regions (level 3), 3883 different households (level 2), and 4064 individuals (level 1).
The table below provides information for every region on the number of households and individuals, and the mean number of individuals per household.

In [9]:
mydata %>%
    group_by(region) %>% 
    summarise(Number_of_households = n_distinct(hid),
             Number_of_individuals = n(),
             Mean_number_of_individuals_per_household = n()/n_distinct(hid))

region Number_of_households Number_of_individuals
1   1     123                  128                  
2   2     435                  454                  
3   3     291                  304                  
4   4     278                  297                  
5   5     262                  279                  
6   6     321                  333                  
7   7     461                  478                  
8   8     458                  483                  
9   9     329                  342                  
10 10     279                  295                  
11 11     369                  384                  
12 12     277                  287                  
   Mean_number_of_individuals_per_household
1  1.040650                                
2  1.043678                                
3  1.044674                                
4  1.068345                                
5  1.064885                                
6  1.037383                                
7  1.036876                                
8  1.054585                                
9  1.039514                                
10 1.057348                                
11 1.040650                                
12 1.036101

- Q2. Specify the generalised equations of a three level multilevel model with random slopes at levels 2 and 3. State the assumptions of the model.

$Y_{ijk} = \beta_0 + \beta_1 x_{1ijk} + v_{0k} + v_{1k}x_{1ijk}   + u_{0jk} + u_{1ijk}x_{1ijk} + e_{ijk}$

Where $Y_{ijk}$ is the expected value of our dependant variable for individual *i* in household *j* in region *k*, $x_{1ijk}$ is a independant variable with random coeffecients at levels two and three, $\beta_0$ is the mean intercept over all slopes, $\beta_1$ is the average slope across all groups (the average in y change across all groups for a 1 unit change in $x_{1ijk}$), and $v_{0k}$,    $v_{1k}x_{1ijk}$,    $u_{0jk}$,    $u_{1ijk}x_{1ijk}$,    $e_{ijk}$ are the random effects associated with level two, level three, and the individual level respectively.

$v_k$ is the effect associated with region *k*, $u_{jk}$ is the effect associated with household *j* within region *k*, and $e_{ijk}$ is the residual error term (the difference between the mean value in household *j*, and the value for individual *i*).


It is assumed that there are cluster effects and supercluster effects

- Q3. Start from the single level null model and add in the household and then region levels. 

A. Display the model coefficients in a table.

B. Does the addition of the household level improve the fit of the model?

C. What evidence is there for an improvement in fit?

#### We start by fitting our null models

First the single level null model.

In [10]:
single_null <- lm(nhood_mistrust ~ 1, data = mydata)
tidy(single_null)

term        estimate std.error  statistic p.value
1 (Intercept) 2.385499 0.01029853 231.6349  0

Then a two level model with household at level 2.

In [11]:
twolevel_null_hid <- lmer(nhood_mistrust ~ (1|hid), data = mydata, REML = FALSE)
summary(twolevel_null_hid)

Linear mixed model fit by maximum likelihood  ['lmerMod']
Formula: nhood_mistrust ~ (1 | hid)
   Data: mydata

     AIC      BIC   logLik deviance df.resid 
  8023.5   8042.4  -4008.7   8017.5     4061 

Scaled residuals: 
    Min      1Q  Median      3Q     Max 
-1.8452 -0.3641 -0.1260  0.3503  2.4936 

Random effects:
 Groups   Name        Variance Std.Dev.
 hid      (Intercept) 0.2622   0.5121  
 Residual             0.1677   0.4095  
Number of obs: 4064, groups:  hid, 3883

Fixed effects:
            Estimate Std. Error t value
(Intercept)  2.38228    0.01046   227.7

We then test for an improvement in fit.

The likelihood ratio test statistic is calculated as two times the difference in thelog likelihood values for the two models.

In [12]:
anova(twolevel_null_hid, single_null)

Df AIC      BIC      logLik    deviance Chisq    Chi Df
single_null       2  8115.927 8128.547 -4055.964 8111.927       NA NA    
twolevel_null_hid 3  8023.475 8042.405 -4008.738 8017.475 94.45193  1    
                  Pr(>Chisq)  
single_null                 NA
twolevel_null_hid 2.511175e-22

In [46]:
qchisq(0.95, 1)

[1] 3.841459

The 

Then a two level model with region at level 2.

In [47]:
twolevel_null_region <- lmer(nhood_mistrust ~ (1|region), data = mydata, REML = FALSE)
summary(twolevel_null_region)

Linear mixed model fit by maximum likelihood . t-tests use Satterthwaite's
  method [lmerModLmerTest]
Formula: nhood_mistrust ~ (1 | region)
   Data: mydata

     AIC      BIC   logLik deviance df.resid 
  8097.7   8116.7  -4045.9   8091.7     4061 

Scaled residuals: 
    Min      1Q  Median      3Q     Max 
-2.2867 -0.5934 -0.1977  0.5542  4.1784 

Random effects:
 Groups   Name        Variance Std.Dev.
 region   (Intercept) 0.003583 0.05986 
 Residual             0.427115 0.65354 
Number of obs: 4064, groups:  region, 12

Fixed effects:
            Estimate Std. Error       df t value Pr(>|t|)    
(Intercept)  2.38271    0.02029 12.51526   117.4   <2e-16 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Then a three level model.

In [48]:
nullmodel_threelvl <- lmer(nhood_mistrust ~ (1|hid) + (1|region), data = mydata, REML = FALSE)
summary(nullmodel_threelvl)

Linear mixed model fit by maximum likelihood . t-tests use Satterthwaite's
  method [lmerModLmerTest]
Formula: nhood_mistrust ~ (1 | hid) + (1 | region)
   Data: mydata

     AIC      BIC   logLik deviance df.resid 
  8005.4   8030.7  -3998.7   7997.4     4060 

Scaled residuals: 
    Min      1Q  Median      3Q     Max 
-1.8163 -0.3721 -0.1164  0.3478  2.6225 

Random effects:
 Groups   Name        Variance Std.Dev.
 hid      (Intercept) 0.258819 0.50874 
 region   (Intercept) 0.003639 0.06032 
 Residual             0.167227 0.40893 
Number of obs: 4064, groups:  hid, 3883; region, 12

Fixed effects:
            Estimate Std. Error       df t value Pr(>|t|)    
(Intercept)  2.37910    0.02049 12.58526   116.1   <2e-16 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

#### We then test to see if there is an improvement 
First we test whether the two-level models are an improvement over the one level models.

In [42]:
anova(twolevel_null_region, single_null)

Df AIC      BIC      logLik    deviance Chisq    Chi Df
single_null          2  8115.927 8128.547 -4055.964 8111.927       NA NA    
twolevel_null_region 3  8097.734 8116.664 -4045.867 8091.734 20.19308  1    
                     Pr(>Chisq)  
single_null                    NA
twolevel_null_region 7.000577e-06

In [42]:
anova(twolevel_null_hid, single_null)

Df AIC      BIC      logLik    deviance Chisq    Chi Df
single_null       2  9299.215 9312.103 -4647.608 9295.215       NA NA    
twolevel_null_hid 3  9193.507 9212.838 -4593.754 9187.507 107.7082  1    
                  Pr(>Chisq)  
single_null                 NA
twolevel_null_hid 3.114109e-25

We then test whether the three-level model is an improvement over the one level model.

In [43]:
anova(nullmodel_threelvl,single_null)

Df AIC      BIC      logLik    deviance Chisq    Chi Df
single_null        2  9299.215 9312.103 -4647.608 9295.215       NA NA    
nullmodel_threelvl 4  9171.232 9197.007 -4581.616 9163.232 131.9835  2    
                   Pr(>Chisq)  
single_null                  NA
nullmodel_threelvl 2.188521e-29

- Q4: Calculate the VPC at the household and regional levels for the 3 level variance components model.

In [44]:
vars_nullmodel1 <- as.data.frame(VarCorr(nullmodel_threelvl))
vars_nullmodel1

grp      var1        var2 vcov        sdcor     
1 hid      (Intercept) NA   0.252559051 0.50255254
2 region   (Intercept) NA   0.003772302 0.06141907
3 Residual NA          NA   0.175598848 0.41904516

In [45]:
region_VPC <- vars_nullmodel1[2,4] / (vars_nullmodel1[1,4] + vars_nullmodel1[2,4] + vars_nullmodel1[3,4])
region_VPC

[1] 0.008733591

In [46]:
hid_VPC <- vars_nullmodel1[1,4] / (vars_nullmodel1[1,4] + vars_nullmodel1[2,4] + vars_nullmodel1[3,4])
hid_VPC

[1] 0.5847219

- Q5. Add in the following explanatory variables in the model (random intercepts model only; no random slopes or coefficients)- age, sclfsato, urban, female, hhtenure, hiqual3. Take out any non-significant associations. 

A. Display the model coefficients in a table.

B. Interpret the coefficients.

Adding in the variables. We have 4 variables at the individual level (age, sclfsato, female, hiqual3) and two variables at the household level (hhtenure, urban).

In [17]:
rndinterceptmodel_threelvl <- lmer(nhood_mistrust ~ agenorm + sclfsato + urban + female +  (1|hid) + (1|region), data = mydata, REML = FALSE)
summary(rndinterceptmodel_threelvl)

Linear mixed model fit by maximum likelihood . t-tests use Satterthwaite's
  method [lmerModLmerTest]
Formula: nhood_mistrust ~ agenorm + sclfsato + urban + female + (1 | hid) +  
    (1 | region)
   Data: mydata

     AIC      BIC   logLik deviance df.resid 
  7646.3   7696.7  -3815.1   7630.3     4056 

Scaled residuals: 
     Min       1Q   Median       3Q      Max 
-2.37016 -0.40100 -0.06812  0.36418  2.85232 

Random effects:
 Groups   Name        Variance Std.Dev.
 hid      (Intercept) 0.229815 0.4794  
 region   (Intercept) 0.001608 0.0401  
 Residual             0.159411 0.3993  
Number of obs: 4064, groups:  hid, 3883; region, 12

Fixed effects:
              Estimate Std. Error         df t value Pr(>|t|)    
(Intercept)  2.720e+00  4.529e-02  7.058e+02  60.050   <2e-16 ***
agenorm     -5.907e-03  5.544e-04  4.059e+03 -10.654   <2e-16 ***
sclfsato    -6.693e-02  6.319e-03  3.800e+03 -10.592   <2e-16 ***
urban        2.550e-01  2.397e-02  2.679e+03  10.639   <2e-16 ***
female 

$Y_{ijk} = \beta_0 + \beta_1 x_{1ijk} + v_{0k} + v_{1k}x_{1ijk}   + u_{0jk} + u_{1ijk}x_{1ijk} + e_{ijk}$

Where $Y_{ijk}$ is the expected value of our dependant variable for individual *i* in household *j* in region *k*, $x_{1ijk}$ is a independant variable with random coeffecients at levels two and three, $\beta_0$ is the mean intercept over all slopes, $\beta_1$ is the average slope across all groups (the average in y change across all groups for a 1 unit change in $x_{1ijk}$), and $v_{0k}$,    $v_{1k}x_{1ijk}$,    $u_{0jk}$,    $u_{1ijk}x_{1ijk}$,    $e_{ijk}$ are the random effects associated with level two, level three, and the individual level respectively.

$v_k$ is the effect associated with region *k*, $u_{jk}$ is the effect associated with household *j* within region *k*, and $e_{ijk}$ is the residual error term (the difference between the mean value in household *j*, and the value for individual *i*).


It is assumed that there are cluster effects and supercluster effects

In [20]:
vcov(rndinterceptmodel_threelvl)

9 x 9 Matrix of class "dpoMatrix"
                  (Intercept)           age      sclfsato         urban
(Intercept)      2.812036e-03 -1.734855e-05 -2.047817e-04 -4.750394e-04
age             -1.734855e-05  3.473374e-07 -1.215023e-07  1.226084e-06
sclfsato        -2.047817e-04 -1.215023e-07  3.968219e-05  1.326986e-06
urban           -4.750394e-04  1.226084e-06  1.326986e-06  5.643858e-04
female          -2.193405e-04  3.654054e-07 -6.560045e-07  1.203827e-05
rent_local_auth -2.743318e-04  2.728740e-06  1.881039e-05 -4.728881e-05
rent_private    -3.924453e-04  4.336020e-06  1.226444e-05 -2.884879e-05
school_qual     -2.928553e-04  1.827777e-07  5.412390e-06 -8.183713e-06
no_qual         -3.041520e-06 -5.256556e-06  4.097818e-06 -1.511796e-05
                       female rent_local_auth  rent_private   school_qual
(Intercept)     -2.193405e-04   -2.743318e-04 -3.924453e-04 -2.928553e-04
age              3.654054e-07    2.728740e-06  4.336020e-06  1.827777e-07
sclfsato        -6.56004

In [25]:
 wald.test(b = coef(rndinterceptmodel_threelvl), Sigma = vcov(rndinterceptmodel_threelvl), Terms = 8:9)

ERROR: Error in `[<-`(`*tmp*`, i, Terms[i], value = 1): subscript out of bounds


- Q5. Add in the following explanatory variables in the model (random intercepts model only; no random slopes or coefficients)- age, sclfsato, urban, female, hhtenure, hiqual3. Take out any non-significant associations. 

A. Display the model coefficients in a table.

B. Interpret the coefficients.

In [48]:
rndinterceptmodel_threelvl2 <- lmer(nhood_mistrust ~ age + sclfsato + urban + hhtenure + hiqual3 + rent_local_auth + rent_private + school_qual + no_qual +(1|hid) + (1|region), data = mydata, REML = FALSE)
summary(rndinterceptmodel_threelvl2)

Linear mixed model fit by maximum likelihood . t-tests use Satterthwaite's
  method [lmerModLmerTest]
Formula: nhood_mistrust ~ age + sclfsato + urban + hhtenure + hiqual3 +  
    (1 | hid) + (1 | region)
   Data: mydata

     AIC      BIC   logLik deviance df.resid 
  7621.0   7677.8  -3801.5   7603.0     4061 

Scaled residuals: 
     Min       1Q   Median       3Q      Max 
-2.30176 -0.40553 -0.06436  0.36998  2.83511 

Random effects:
 Groups   Name        Variance Std.Dev.
 hid      (Intercept) 0.225538 0.47491 
 region   (Intercept) 0.001461 0.03822 
 Residual             0.159942 0.39993 
Number of obs: 4070, groups:  hid, 3888; region, 12

Fixed effects:
              Estimate Std. Error         df t value Pr(>|t|)    
(Intercept)  2.564e+00  5.969e-02  1.729e+03  42.959  < 2e-16 ***
age         -5.321e-03  5.771e-04  4.054e+03  -9.219  < 2e-16 ***
sclfsato    -6.194e-02  6.320e-03  3.810e+03  -9.800  < 2e-16 ***
urban        2.439e-01  2.386e-02  2.564e+03  10.223  < 2e-16 ***

In [21]:
model1 <- lmer(nhood_mistrust ~ (1|hid) + (1|region), data = mydata, REML = FALSE)
model2 <- lmer(nhood_mistrust ~ agenorm + (1|hid) + (1|region), data = mydata, REML = FALSE)
anova(model1, model2)

Df AIC      BIC      logLik    deviance Chisq    Chi Df Pr(>Chisq)  
model1 4  8005.431 8030.670 -3998.715 7997.431       NA NA               NA
model2 5  7862.912 7894.462 -3926.456 7852.912 144.5185  1     2.736695e-33

In [22]:
model3 <- lmer(nhood_mistrust ~ agenorm + sclfsato + (1|hid) + (1|region), data = mydata, REML = FALSE)
anova(model2, model3)

Df AIC      BIC      logLik    deviance Chisq    Chi Df Pr(>Chisq)  
model2 5  7862.912 7894.462 -3926.456 7852.912       NA NA               NA
model3 6  7753.383 7791.242 -3870.691 7741.383 111.5295  1     4.530027e-26

In [23]:
model4 <- lmer(nhood_mistrust ~ agenorm + sclfsato + urban + (1|hid) + (1|region), data = mydata, REML = FALSE)
anova(model3, model4)

Df AIC      BIC      logLik    deviance Chisq    Chi Df Pr(>Chisq)  
model3 6  7753.383 7791.242 -3870.691 7741.383       NA NA               NA
model4 7  7644.384 7688.554 -3815.192 7630.384 110.9984  1     5.921699e-26

In [24]:
model5 <- lmer(nhood_mistrust ~ agenorm + sclfsato + urban + school_qual + no_qual + (1|hid) + (1|region), data = mydata, REML = FALSE)
anova(model4, model5)

Df AIC      BIC      logLik    deviance Chisq    Chi Df Pr(>Chisq)  
model4 7  7644.384 7688.554 -3815.192 7630.384       NA NA               NA
model5 9  7629.030 7685.819 -3805.515 7611.030 19.35401  2     6.270894e-05

In [25]:
model6 <- lmer(nhood_mistrust ~ agenorm + sclfsato + urban + school_qual + no_qual + rent_local_auth + rent_private + (1|hid) + (1|region), data = mydata, REML = FALSE)
anova(model5, model6)

Df AIC      BIC      logLik    deviance Chisq    Chi Df Pr(>Chisq)  
model5  9 7629.030 7685.819 -3805.515 7611.030       NA NA               NA
model6 11 7558.826 7628.235 -3768.413 7536.826 74.20383  2     7.706262e-17

In [26]:
model7 <- lmer(nhood_mistrust ~ agenorm + sclfsato + urban + school_qual + no_qual + rent_local_auth + rent_private + female +(1|hid) + (1|region), data = mydata, REML = FALSE)
anova(model6, model7)

Df AIC      BIC      logLik    deviance Chisq     Chi Df Pr(>Chisq)
model6 11 7558.826 7628.235 -3768.413 7536.826        NA NA            NA 
model7 12 7560.595 7636.314 -3768.298 7536.595 0.2309644  1     0.6308097

- Q6. Add in random slopes for age, at the household and regional level

A. Does this improve the fit of the model?

B. Take out from the model any non-significant random slope(s) for age.

C. Interpret the random slope(s) for age, with the help of graphs.

In [18]:
rndslopemodel_threelvl <- lmer(nhood_mistrust ~ age + sclfsato + urban + female + hhtenure + hiqual3 + (1 + age|hid) + (1 + age|region), data = mydata, REML = FALSE, control = lmerControl(check.nobs.vs.nRE = "ignore"))
summary(rndslopemodel_threelvl)

Warning message in checkConv(attr(opt, "derivs"), opt$par, ctrl = control$checkConv, :
“unable to evaluate scaled gradient”Warning message in checkConv(attr(opt, "derivs"), opt$par, ctrl = control$checkConv, :
“Model failed to converge: degenerate  Hessian with 1 negative eigenvalues”Warning message:
“Model failed to converge with 1 negative eigenvalue: -5.6e+03”

Linear mixed model fit by maximum likelihood . t-tests use Satterthwaite's
  method [lmerModLmerTest]
Formula: nhood_mistrust ~ age + sclfsato + urban + female + hhtenure +  
    hiqual3 + (1 + age | hid) + (1 + age | region)
   Data: mydata
Control: lmerControl(check.nobs.vs.nRE = "ignore")

     AIC      BIC   logLik deviance df.resid 
  7573.8   7662.2  -3772.9   7545.8     4050 

Scaled residuals: 
     Min       1Q   Median       3Q      Max 
-2.56480 -0.35865 -0.06094  0.34355  3.08626 

Random effects:
 Groups   Name        Variance  Std.Dev. Corr 
 hid      (Intercept) 5.478e-01 0.740119      
          age         6.723e-05 0.008200 -0.81
 region   (Intercept) 1.500e-04 0.012247      
          age         1.143e-06 0.001069 -0.94
 Residual             1.326e-01 0.364211      
Number of obs: 4064, groups:  hid, 3883; region, 12

Fixed effects:
              Estimate Std. Error         df t value Pr(>|t|)    
(Intercept)  2.571e+00  5.959e-02  2.767e+03  43.148  < 2e-16 ***
age

- Q7. Examine potential interaction effects between age and the other explanatory variables- interpret the interactions using graphs. Is the random slope of age is explained by these interaction terms?

- Q8. After fitting all the explanatory variables, is a three level model still appropriate?

- Q9. Display and interpret the parameters in your final model in a table.

- Q10. Examine residuals at all three levels- are the assumptions of the regression model met?

#### Part B: Dependent variable: worry_crime

Fit a multilevel logistic model with worry_crime as the dependent variable.

- Q11. Start from the single level null model and add in the household and then GOR levels. Display the model coefficients for the 3 level variance components model  in a table.

- Q12. Is a three level model appropriate? Is a 2 level model appropriate? Is a single level model appropriate? If a 2 level model is appropriate, which two level models (individuals within HH or individuals within GOR)? State your reasons for choosing between a 2 vs 3 level model.

- Q13. Having decided whether a multilevel or single level model is appropriate, add in the explanatory variables from your final model in Part A. Compare the models using a Table. Are the associations with worry_crime similar to the associations with nhood_mistrust?

- Q14. Take out any non-significant explanatory variables from the model. Interpret the coefficients in your final model in a table.